src/main/scala/chiseltest/legacy/backends/verilator/VerilatorExecutive.scala

```diff
@@ -48,6 +52,74 @@ object VerilatorExecutive extends BackendExecutive {
     val circuit = elaboratedAnno.collect { case x: ChiselCircuitAnnotation => x }.head.circuit
     val dut = getTopModule(circuit).asInstanceOf[T]
 
+    if (elaboratedAnno.contains(CachingAnnotation)) {
+      // Generate new high FIRRTL for caching
+      val highFirrtlPath = s"${targetDir}/${circuit.name}.hi.fir"
+      val oldHighFirrtlPath = s"${targetDir}/${circuit.name}.hi.fir.old"
+      s"mv ${highFirrtlPath} ${oldHighFirrtlPath}".!
+
+      (new ChiselStage).run(
+        elaboratedAnno :+ RunFirrtlTransformAnnotation(new HighFirrtlEmitter)
+      )
+
+      // Diff high FIRRTL and delete old file
+      val diffExit = s"diff ${highFirrtlPath} ${oldHighFirrtlPath}".!
```

Comment:
Instead of relying on GNU tools here, you could try to use the hashing mechanism that we already have for firrtl:
https://github.com/chipsalliance/firrtl/blob/edb91f7bc613026f824519786c3ce25740bb21c3/src/main/scala/firrtl/ir/StructuralHash.scala#L29

Reply:
Changed to use your structural hash (thank you!) and replaced Unix operations in general with native Scala / Java file operations.